benchmarks/geometry_providers.jl

```diff
@@ -0,0 +1,181 @@
+#=
+# Geometry providers
+
+This file benchmarks GeometryOps methods on every GeoInterface.jl implementation we can find, in order to test:
+a. genericness, i.e., does GeometryOps work correctly with all GeoInterface.jl implementations?
+b. performance, i.e., how does GeometryOps compare to the native implementation?
+c. performance issues in the packages' implementations of GeoInterface
+=#
+
+# First, we import the providers:
+using ArchGDAL, LibGEOS, Shapefile, GeoJSON, WellKnownGeometry, GeometryBasics, GeoInterface
+
+import GeometryOps as GO, GeoInterface as GI
+
+using BenchmarkTools, Chairmarks, CairoMakie, MakieThemes, DataFrames, Proj
+
+PROVIDERS = (ArchGDAL, LibGEOS, GeometryBasics, GI.Wrappers)
+
+
+import Polylabel
+
+water1 = ArchGDAL.fromWKT([readchomp(joinpath(dirname(dirname(pathof(Polylabel))), "test", "data", "water1.wkt")) |> String])
+water2 = ArchGDAL.fromWKT([readchomp(joinpath(dirname(dirname(pathof(Polylabel))), "test", "data", "water2.wkt")) |> String])
+
+water1_1 = GO.fix(water1, corrections = [GO.GEOSCorrection()])
+water2 = GO.fix(water2, corrections = [GO.GEOSCorrection()])
+
+using CoordinateTransformations, Rotations
+w1rg = GO.transform(Translation(GO.centroid(water1)) ∘ LinearMap(Makie.rotmatrix2d(π/2)) ∘ Translation((-).(GO.centroid(water1))), water1)
```

Comment:
It would be nice to have some comments on what these different geometries are for.

Reply:
Just more comments in general in the benchmark files about what the different sections are doing would be nice as they are quite dense.

Reply:
Will do!  They could definitely use some cleanup as well.